core/contracts/financial_templates/Liquidatable.sol

```diff
@@ -240,35 +249,30 @@ contract Liquidatable is PricelessPositionManager {
      * that the DVM has resolved a price. This method then calculates whether the
      * dispute on the liquidation was successful usin only the settlement price,
      * tokens outstanding, locked collateral (post-pending withdrawals), and liquidation ratio
-     *
-     * TODO: Requests a settlement price from the DVM
-     * TESTING: For now, I allow the caller to hard-code a settlement price and
-     * a dispute resolution => {SUCCESS, FAILURE}
      */
-    function settleDispute(uint id, address sponsor, FixedPoint.Unsigned memory hardcodedPrice, bool disputeSucceeded)
-        public
-        onlyPendingDispute(id, sponsor)
-    {
+    function settleDispute(uint id, address sponsor) public onlyPendingDispute(id, sponsor) {
         LiquidationData storage disputedLiquidation = _getLiquidationData(sponsor, id);
 
-        // if (disputedLiquidation.oracle.hasPrice(disputedLiquidation.identifier, disputedLiquidation.disputeTime)) {
-        // If dispute is over set oracle price
-        // disputedLiquidation.oraclePrice = disputedLiquidation.oracle.getPrice(
-        //     disputedLiquidation.identifier,
-        //     disputedLiquidation.disputeTime
-        // );
+        // Get the returned price from the oracle. If this has not yet resolved will revert.
+        disputedLiquidation.settlementPrice = _getOraclePrice(disputedLiquidation.disputeTime);
+
+        // Find the value of the tokens in the underlying collateral.
+        FixedPoint.Unsigned memory tokenRedemptionValue = disputedLiquidation.tokensOutstanding.mul(
+            disputedLiquidation.settlementPrice
+        );
+
+        // The required collateral is the value of the tokens in underlying * required collateral ratio.
+        FixedPoint.Unsigned memory requiredCollateral = tokenRedemptionValue.mul(collateralRequirement);
 
-        // TODO: For testing purposes
-        disputedLiquidation.settlementPrice = hardcodedPrice;
+        // If the position has more than the required collateral it is solvent and the dispute is valid(liquidation is invalid)
+        // Note that this check uses the liquidatedCollateral not the lockedCollateral as this considers withdrawals.
+        // bool disputeSucceeded = !requiredCollateral.isLessThan(disputedLiquidation.liquidatedCollateral);
+        bool disputeSucceeded = disputedLiquidation.liquidatedCollateral.isGreaterThan(requiredCollateral);
```

Comment:
nit: this should be greater than or equal to.

Reply:
it should indeed. I tried to implement this using `!isLessThan` but had issued with this. will park this for now until we implement a `isGreatThanOrEqual` in the `FixedPoint` lib.

Reply:
I will try again with the `!isLessThan` notation and see how it goes.